packages/web-pkg/src/composables/actions/spaces/useSpaceActionsDuplicate.ts

```diff
@@ -0,0 +1,124 @@
+import { SpaceResource } from '@ownclouders/web-client'
+import { computed } from 'vue'
+import { SpaceAction, SpaceActionOptions } from '../types'
+import { useGettext } from 'vue3-gettext'
+import { useStore } from '../../store'
+import { useAbility } from '../../ability'
+import { useClientService } from '../../clientService'
+import { useLoadingService } from '../../loadingService'
+import { Store } from 'vuex'
+import { buildSpace, isProjectSpaceResource } from '@ownclouders/web-client/src/helpers'
+import { Drive } from '@ownclouders/web-client/src/generated'
+import { resolveFileNameDuplicate } from '../../../helpers'
+
+export const useSpaceActionsDuplicate = ({
+  store
+}: {
+  store?: Store<any>
+} = {}) => {
+  store = store || useStore()
+  const { $gettext } = useGettext()
+  const ability = useAbility()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+
+  const duplicateSpace = async (spaces: SpaceResource[]) => {
+    const existingSpace = spaces[0]
+    console.log(existingSpace)
+    const projectSpaces: SpaceResource[] = store.getters['runtime/spaces/spaces'].filter(
+      (space: SpaceResource) => isProjectSpaceResource(space)
+    )
+    let newSpace: SpaceResource = null
+    const newSpaceName = resolveFileNameDuplicate(existingSpace.name, '', projectSpaces)
+
+    try {
+      const { data: createdSpace } = await clientService.graphAuthenticated.drives.createDrive(
+        {
+          name: newSpaceName,
+          description: existingSpace.description
+        },
+        {}
+      )
+      newSpace = buildSpace(createdSpace)
+
+      const existingSpaceFiles = await clientService.webdav.listFiles(existingSpace)
+      for (const file of existingSpaceFiles.children) {
+        await clientService.webdav.copyFiles(existingSpace, file, newSpace, { path: file.name })
```

Comment:
Does this need to happen synchronously, or could you resolve the promises asynchronously via `Promise.all(promises)`?